cpp/src/arrow/compute/kernels/scalar_set_lookup_test.cc

```diff
@@ -72,6 +72,27 @@ void CheckIsInChunked(const std::shared_ptr<ChunkedArray>& input,
   AssertChunkedEqual(*expected, *actual);
 }
 
+void CheckIsInDictionary(const std::shared_ptr<DataType>& type,
+                         const std::shared_ptr<DataType>& index_type,
+                         const std::string& input_dictionary_json,
+                         const std::string& input_index_json,
+                         const std::string& value_set_json,
+                         const std::string& expected_json, bool skip_nulls = false) {
+  auto dict_type = dictionary(index_type, type);
+  auto indices = ArrayFromJSON(index_type, input_index_json);
+  auto dict = ArrayFromJSON(type, value_set_json);
```

Comment:
This should use `input_dictionary_json`, not `value_set_json`.

Reply:
Ugh, thanks for catching that. Fixed.